code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,558 @@
+
+	if(IS_OPAQUE_TURF(T))
+		if(!silent)
+			to_chat(owner, "<span class='xenowarning'>We can't blink into this space without vision!</span>")
+		return FALSE
+
+	for(var/atom/blocker as() in T)
+		if(blocker.CanPass(owner, T))
+			continue
+
+		if(!silent)
+			to_chat(owner, "<span class='xenowarning'>We can't blink into a solid object!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/activable/blink/use_ability(atom/A)
+	. = ..()
+	var/mob/living/carbon/xenomorph/wraith/X = owner
+	var/turf/T = X.loc
+	var/turf/temp = X.loc
+
+	for (var/x = 1 to WRAITH_BLINK_RANGE)
+		temp = get_step(T, get_dir(T, A))
+		if (!temp)
+			break
+		T = temp
+
+	if(!can_use_ability(T)) //Since we updated the turf, check it again.
+		return fail_activate()
+
+	X.face_atom(T) //Face the target so we don't look like an ass
+
+	teleport_debuff_aoe(X) //Debuff when we vanish
+
+	X.forceMove(T) //Teleport to our target turf
+
+	var/cooldown_mod = 1
+	var/pulled_target = owner.pulling
+	if(pulled_target) //bring the pulled target with us if applicable but at the cost of sharply increasing the next cooldown
+		cooldown_mod = WRAITH_BLINK_DRAG_MULTIPLIER
+		to_chat(X, "<span class='xenodanger'>We bring [pulled_target] with us. We won't be ready to blink again for [cooldown_timer * 0.1] seconds due to the strain of doing so.</span>")
+
+	teleport_debuff_aoe(X) //Debuff when we reappear
+
+	succeed_activate()
+	add_cooldown(cooldown_mod)
+
+	GLOB.round_statistics.wraith_blinks++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_blinks") //Statistics
+
+///Called by many of the Wraith's teleportation effects
+/datum/action/xeno_action/proc/teleport_debuff_aoe(atom/movable/teleporter, silent = FALSE)
+	var/mob/living/carbon/xenomorph/ghost = owner
+
+	if(!silent) //Sound effects
+		playsound(teleporter, 'sound/effects/EMPulse.ogg', 25, 1) //Sound at the location we are arriving at
+
+	new /obj/effect/temp_visual/blink_portal(get_turf(teleporter))
+
+	for(var/turf/affected_tile as() in RANGE_TURFS(1,teleporter.loc))
+		affected_tile.add_filter("wraith_blink_distortion", 3, list("type" = "motion_blur", 0, 0)) //Cool filter appear
+		animate(affected_tile.get_filter("wraith_blink_distortion"), x = 60*rand() - 30, y = 60*rand() - 30, time = 0.5 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+		addtimer(CALLBACK(affected_tile, /atom.proc/remove_filter, "wraith_blink_distortion"), 1 SECONDS)
+
+		for(var/obj/obj_target in affected_tile) //This is just about SFX, so we don't have objects not distorting while everything else does
+			obj_target.add_filter("wraith_aoe_debuff_filter", 3, list("type" = "motion_blur", 0, 0)) //Cool filter appear
+			animate(obj_target.get_filter("wraith_aoe_debuff_filter"), x = 60*rand() - 30, y = 60*rand() - 30, time = 0.25 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+			addtimer(CALLBACK(obj_target, /atom.proc/remove_filter, "wraith_aoe_debuff_filter"), 0.5 SECONDS)
+
+		for(var/mob/living/living_target in affected_tile)
```

Comment:
since the code for those two is basically same, why not for(var/atom/movable in affected_tile) ?
then do a isliving() for the rest of living code

Reply:
this comment is not responded to

Reply:
Already tried; it totally fucks with the visuals

Reply:
this is fine as is for now.